### PR TITLE
Add tests coverage for PhotoRequester

### PR DESCRIPTION
**Pull request by EdenKahane**
PhotoRequester is a hard class to test because it relies on permissions and ActivityResultContracts.
The best way to test that I found is following this : https://medium.com/@adrian.gl/how-to-mock-and-test-activityresult-apis-with-jetpack-compose-f3a1d6311171
But I added more type check safety because otherwise finer testing was not possible.

### CODE REVIEW

File: app/src/androidTest/java/com/android/bookswap/model/PhotoRequesterTest.kt [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+import java.io.File
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertThrows
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+/*
+Thanks https://medium.com/@adrian.gl/how-to-mock-and-test-activityresult-apis-with-jetpack-compose-f3a1d6311171
```
Comment (matiisalv) [inline app/src/androidTest/java/com/android/bookswap/model/PhotoRequesterTest.kt:34]: could add explanation about it  in DocString

**Review by matiisalv — CHANGES_REQUESTED** [review]
Amazing work as usual , but could a comment in the rest  of the code , because give explanation in a certain part ,other than that it 's perfect, good work .

File: app/src/androidTest/java/com/android/bookswap/model/PhotoRequesterTest.kt (hunk 1 above)
Comment (EdenKahane) [reply app/src/androidTest/java/com/android/bookswap/model/PhotoRequesterTest.kt:34]: The function that I documented is a complicated helper function. But this is just a test class like all the other so the documentation is not needed.

**Review by matiisalv — APPROVED** [review]
as we discuss all good

**Pull request merged**